### PR TITLE
Revise the text on ticket age handling on the client and server. Fixe…

### PR DESCRIPTION
**Pull request by ekr**
…s #919, #940, #944.#944

### CODE REVIEW

File: draft-ietf-tls-tls13.md [hunk 1]
```diff
@@ -2684,6 +2680,22 @@ facilitates implementation as described below). Servers MUST check
...
 
+#### Ticket Age
+
+The client's view of the age of a ticket is the time since the receipt
+of the NewSessionTicket message. Clients MUST NOT attempt to use
+tickets which have ages greater than the "ticket_lifetime" value which
+was provided with the ticket. The "obfuscated_ticket_age" field of
+each PskIdentity contains an obfuscated version of the ticket age
+formed by taking the age in milliseconds and adding the "ticket_age_add"
+value that was included with the ticket, see {{NSTMessage}} module 2^32.
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: modulo

**Comment by kazu-yamamoto** [comment]
👍

File: draft-ietf-tls-tls13.md [hunk 2]
```diff
@@ -2735,50 +2747,55 @@ EndOfEarlyData message.
...
+  from the operating system.
+- The "ticket_age_add" parameter from the NewSessionTicket message in
+  which the ticket was established.
+
+The server can determine the client's view of the age of the ticket by
+subtracting the ticket's "ticket_age_add value" from the
+"obfuscated_ticket_age" parameter in the client's "pre_shared_key"
+extension. The server can independently determine its view of the
+age of the ticket by subtracting the the time the ticket was issued
+from the curren time. If the client and server clocks were running
```
Comment (kaduk) [inline draft-ietf-tls-tls13.md]: "current"

File: draft-ietf-tls-tls13.md [hunk 3]
```diff
@@ -2735,50 +2747,55 @@ EndOfEarlyData message.
...
-symmetric, so the optimal tradeoff may involve an asymmetric replay window.
-
+measurement of time difficult. Variations in client and server clock
+rates are likely to be minimal, though potentially with gross time
+corrections.  Network propagation delays are the most likely causes of
+a mismatch in legitimate values for elapsed time.  Both the
+NewSessionTicket and ClientHello messages might be retransmitted and
+therefore delayed, which might be hidden by TCP. Thus, an
+an allowance on the order of ten seconds to account for errors in clocks and
+variations in measurements is advisable. Outside this range, the
```
Comment (kaduk) [inline draft-ietf-tls-tls13.md]: When I wrote the original text, I was careful to scope the numerical guidance to the scenario used to compute it.  It's unclear that 10 seconds is still good for the gamut from IoT to intra-datacenter to space missions.  The RTT measurement helps some, but is not a guarantee.

**Comment by ekr** [comment]
@kaduk: if you want to propose some revised text, I will TAL.

**Pull request merged**